urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/net/contrapunctus/lzma/LzmaOutputStream.java

```diff
@@ -0,0 +1,100 @@
+// LzmaOutputStream.java -- transparently compress LZMA while writing
+// Copyright (c)2007 Christopher League <league@contrapunctus.net>
+
+// This is free software, but it comes with ABSOLUTELY NO WARRANTY.
+// GNU Lesser General Public License 2.1 or Common Public License 1.0
+
+package net.contrapunctus.lzma;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+
+public class LzmaOutputStream extends FilterOutputStream
+{
+    protected EncoderThread          eth;
+    
+    public static boolean            LZMA_HEADER = true;
+    /*
+     * true for compatibility with lzma(1) command-line tool, false for
+     * compatibility with previous versions of LZMA streams.
+     */
+    
+    private static final PrintStream dbg         = System.err;
+    private static final boolean     DEBUG;
+    
+    static
+    {
+        String ds = null;
+        try
+        {
+            ds = System.getProperty("DEBUG_LzmaStreams");
+        }
+        catch (final SecurityException e)
+        {
+        }
+        DEBUG = ds != null;
+    }
+    
+    public LzmaOutputStream(OutputStream _out)
+    {
+        this(_out, EncoderThread.DEFAULT_DICT_SZ_POW2, null);
+    }
+    
+    public LzmaOutputStream(OutputStream _out, Integer dictSzPow2,
+            Integer fastBytes)
+    {
+        super(null);
+        eth = new EncoderThread(_out, dictSzPow2, fastBytes);
+        out = ConcurrentBufferOutputStream.create(eth.q);
+        if (DEBUG)
+        {
+            dbg.printf("%s >> %s (%s)%n", this, out, eth.q);
+        }
+        eth.start();
+    }
+    
+    
+    public void write(int i) throws IOException
+    {
+        if (eth.exn != null)
+        {
+            throw eth.exn;
+        }
+        out.write(i);
+    }
+    
+    
+    public void close() throws IOException
+    {
+        if (DEBUG)
+        {
+            dbg.printf("%s closed%n", this);
+        }
+        out.close();
+        try
+        {
+            eth.join();
+            if (DEBUG)
+            {
+                dbg.printf("%s joined %s%n", this, eth);
+            }
+        }
+        catch (final InterruptedException exn)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)